extensions/libarchive/CompressContent.h

```diff
@@ -190,16 +189,11 @@ class CompressContent: public core::Processor {
       return ret;
     }
 
-    static la_ssize_t archive_read(struct archive *arch, void *context, const void **buff) {
-      WriteCallback *callback = (WriteCallback *) context;
+    static la_ssize_t archive_read(struct archive*, void *context, const void **buff) {
+      auto *callback = (WriteCallback *) context;
       callback->session_->read(callback->flow_, &callback->readDecompressCb_);
-      if (callback->readDecompressCb_.read_size_ >= 0) {
-        *buff = callback->readDecompressCb_.buffer_;
-        return gsl::narrow<la_ssize_t>(callback->readDecompressCb_.read_size_);
-      } else {
-        archive_set_error(arch, EIO, "Error reading flowfile");
-        return -1;
-      }
+      *buff = callback->readDecompressCb_.buffer_;
+      return gsl::narrow<la_ssize_t>(callback->readDecompressCb_.read_size_);
```

Comment:
[this](http://manpages.org/archive_read_open2/3) I-don't-know-how-trustworthy site says:

> On error, the read callback should invoke Fn archive_set_error to register an error code and message and return -1.

Reply:
Good catch. The site lists the official man page contents in a web format, so I would say the contents are valid. I've mistakenly removed the error branch because after converting `read_size_` to `size_t`, it was a dead branch and I didn't realize `read_size_` comes from a stream read return value. Will fix and rename `read_size_` to `stream_read_result`.

Reply:
fixed in 7955d00

Reply:
`archive_set_error` should also be restored according to the link

Reply:
restored `archive_set_error` in e8a0a33